cmake/add_module.cmake

```diff
@@ -0,0 +1,47 @@
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+
+include(CMakeParseArguments)
+
+function (add_module)
+
+  cmake_parse_arguments(ADD_MODULE "" "TARGET" "GRAMMAR;SOURCES;DEPENDS;INCLUDES" ${ARGN})
+
+  generate_y_from_ym("modules/${ADD_MODULE_TARGET}/${ADD_MODULE_GRAMMAR}")
+  bison_target(${ADD_MODULE_TARGET}Grammar}
+    ${CMAKE_CURRENT_BINARY_DIR}/${ADD_MODULE_GRAMMAR}.y
+    ${CMAKE_CURRENT_BINARY_DIR}/${ADD_MODULE_GRAMMAR}.c
+    COMPILE_FLAGS ${BISON_FLAGS}
+    )
+
+  include_directories(${ADD_MODULE_INCLUDES})
```

Comment:
This is not okay, it should be added to the `target_include_directories`

Reply:
Well.. this is another pattern that was accepted until now ;)